### PR TITLE
feat: AbsoluteQuantitationMethodFile::store()

### PR DESCRIPTION
**Pull request by pcolaianni**
implementation proposal

the PR also contains my modification from another branch (the one which fixes the AQM class), the only interesting thing here is the `store()` method.

here: https://github.com/biosustain/OpenMS/pull/99/files#diff-de82869c936b9970c20ccdc4f9d9bd61R213

the modification on how the map is used (eg. using `.at()`) in `load()` were necessary to enforce `const` correctness (due to changes in AQM class from the other branch).

### CODE REVIEW

File: src/openms/source/FORMAT/AbsoluteQuantitationMethodFile.cpp [hunk 1]
```diff
@@ -216,9 +210,58 @@ namespace OpenMS
     aqm.setTransformationModelParams(transformation_model_params);
   }
 
-  // void AbsoluteQuantitationMethodFile::store(const String & filename, const std::vector<AbsoluteQuantitationMethod> & aqm_list)
-  // {
-  //   // TODO: pending fix to CsvFile::fstore()
-  // }
-
+  void AbsoluteQuantitationMethodFile::store(
```
Comment (dmccloskey) [inline src/openms/source/FORMAT/AbsoluteQuantitationMethodFile.cpp:213]: Was there a reason that you chose not to use the Csv::addRow and Csv::store methods?

Comment (pcolaianni) [reply src/openms/source/FORMAT/AbsoluteQuantitationMethodFile.cpp:213]: I could convert each aqm to a StringList and then take advantage of those methods, yeah. Did not think about it. I'll try that solution as well and I'll see if there is anything wrong with it.

**Comment by pcolaianni** [comment]
@dmccloskey The PR is updated.

File: src/openms/source/FORMAT/AbsoluteQuantitationMethodFile.cpp [hunk 2]
```diff
@@ -213,55 +213,50 @@ namespace OpenMS
...
   {
-    std::ofstream ofs(filename);
-    if (!ofs.is_open())
-    {
-      throw Exception::UnableToCreateFile(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, filename);
-    }
-    std::vector<String> tm_params_names;
+    clear(); // clear the buffer_
+    const String headers = "IS_name,component_name,feature_name,concentration_units,llod,ulod,lloq,uloq,correlation_coefficient,n_points,transformation_model";
+    StringList splitted_headers;
```
Comment (dmccloskey) [inline src/openms/source/FORMAT/AbsoluteQuantitationMethodFile.cpp]: Might want to rename "splitted_headers" to "split_headers"

File: src/openms/source/FORMAT/AbsoluteQuantitationMethodFile.cpp [hunk 3]
```diff
@@ -213,55 +213,50 @@ namespace OpenMS
...
-    // Assuming the following headers are always present
-    ofs << "IS_name,component_name,feature_name,concentration_units,llod,ulod,lloq,uloq,correlation_coefficient,n_points,transformation_model";
-    // However, the following headers could be absent
-    for (const String tm_param_header : tm_params_names)
-    {
-      ofs << "," << "transformation_model_param_" << tm_param_header;
-    }
-    ofs << std::endl;
-    // Write raw data
+    addRow(splitted_headers);
```
Comment (dmccloskey) [inline src/openms/source/FORMAT/AbsoluteQuantitationMethodFile.cpp]: Think this implementation looks much cleaner.

**Review by dmccloskey — CHANGES_REQUESTED** [review]
I think it looks much cleaner using the addRow and store methods you added to Csv

**Comment by pcolaianni** [comment]
Yeah it was the way to go. I had forgotten about `addRow()`, even though I had implemented it recently. :sweat_smile: 
Thanks for your reviews!

**Pull request closed without merging**